Refactor dev requirements out of local

jgmize opened the pull request:
@jwhitlock r?

jwhitlock commented on requirements/dev.txt:
```diff
@@ -0,0 +1,47 @@
+tox==2.3.1 \
+    --hash=sha256:1823c93ca378092c10bbde428213d3f5066b30adb09e5c001087a83e3e0a402a \
+    --hash=sha256:bf7fcc140863820700d3ccd65b33820ba747b61c5fe4e2b91bb8c64cb21a47ee
+# Enables ./manage.py runserver_plus with in-browser exception debugging
```
Blank space between ``tox`` and ``Werkzeug`` comment please.

jwhitlock commented:
TravisCI is happy, and the requirements that moved look good to me.

Add ``dev.txt`` line to ``requirements/README.rst``, and update the Future if you are bold enough.

jwhitlock commented:
Also, prefix commits with ``bug 1218563:``

jgmize commented:
Thanks for the quick feedback. I added the missing space, updated requirements/README.rst, and prefixed the commit message with the bug #. I went ahead and amended the commit this time as the changes were only docs and whitespace, but let me know if you'd prefer to always have separate commits and squash only after r+ and I'll stick to that in the future.

jwhitlock commented:
👍 I prefer amended, rebased commits. I find them easier to read 6+ months later, and GitHub support for them gets better every update.
